WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -376,16 +385,7 @@ public void onClick(View view) {
 
         mStickySwitch.setOnCheckedChangeListener(mOnStickySwitchChangeListener);
 
-
-        if (getEditPostRepository() != null && getEditPostRepository().isPage()) { // remove post specific views
-            final View categoriesTagsContainer = rootView.findViewById(R.id.post_categories_and_tags_card);
-            final View formatBottomSeparator = rootView.findViewById(R.id.post_format_bottom_separator);
-            final View markAsStickyContainer = rootView.findViewById(R.id.post_settings_mark_as_sticky_container);
-            categoriesTagsContainer.setVisibility(View.GONE);
-            formatBottomSeparator.setVisibility(View.GONE);
-            mFormatContainer.setVisibility(View.GONE);
-            markAsStickyContainer.setVisibility(View.GONE);
-        }
+        if (getEditPostRepository() != null) hideSpecificViews(rootView, getEditPostRepository().isPage());
```

Comment:
Yeah, that was my first thought as well. I was just worried that since this is being called in the `onCreateView` method, having it depend on fields would make it implicitly dependent on **where** that call should happen, specially when called inside `onCreateView`, and changing the call position inside the method could cause a crash.

For instance:
- Calling the method at the end of `onCreateView` would likely always work as those fields are being defined in the body of that method, prior to the call.
- But if for some reason (refactor or something) that call is moved up to the start of the `onCreateView` body, it would mean those variables are still not declared (`null`) which would either cause a crash or cause the function to not work as expected.

Reply:
I understand your point @thomashorta 👍 

Seems this is a common problem mostly left unsolved Android (views) code.

Thank you for the explanation 🙇

Reply:
I ended up doing what you suggested and moving the call to `hideSpecificViews` to the bottom of the `onCreateView` method. It makes it look better by just accessing fields, we just have to be careful about call site, but should be fine.